src/Build/Logging/FancyLogger/FancyLogger.cs

```diff
@@ -44,11 +45,48 @@ public void Initialize(IEventSource eventSource)
             eventSource.ErrorRaised += new BuildErrorEventHandler(eventSource_ErrorRaised);
             // Cancelled
             Console.CancelKeyPress += new ConsoleCancelEventHandler(console_CancelKeyPressed);
-            // Initialize FancyLoggerBuffer
-            FancyLoggerBuffer.Initialize();
-            // TODO: Fix. First line does not appear at top. Leaving empty line for now
-            FancyLoggerBuffer.WriteNewLine(string.Empty);
-            FancyLoggerBuffer.Render();
+            
+            Task.Run(() =>
+            {
+                // Initialize FancyLoggerBuffer
+                FancyLoggerBuffer.Initialize();
+                // TODO: Fix. First line does not appear at top. Leaving empty line for now
+                FancyLoggerBuffer.WriteNewLine(string.Empty);
+                // First render
+                FancyLoggerBuffer.Render();
+                int i = 0;
+                // Rerender periodically
+                while (!FancyLoggerBuffer.IsTerminated)
+                {
+                    i++;
+                    // Delay by 1/60 seconds
+                    Task.Delay((i / 60) * 1_000).ContinueWith((t) =>
```

Comment:
ContinueWith is a bit surprising here. Is there a straightforward `async`/`await` implementation, or can you comment the need for this more advanced mechanism?

Reply:
Ah, not new to this PR so I still want to know about this but not blocking.

Reply:
Yes. AFAIK `await` pauses the execution of the task meaning that keyboard input would also be delayed. This way it is deferred (hence the i) so the input can be handled with no delay.
I might be wrong though, I am thinking of this similar to javascript promises.